### PR TITLE
Create introduction/what-is.md

### PR DESCRIPTION
**Pull request by rrrutledge**
This file holds the text to accompany the [What is InnerSource?](https://www.safaribooksonline.com/videos/introduction-to-innersource/9781492041504/9781492041504-video321608) video.
This accompanying text will appear as a part of the learning path (like [this example](https://www.safaribooksonline.com/learning-paths/learning-path-lean/9781491999738/9781491946527-/part01ch01.html)).

### CODE REVIEW

File: introduction/what-is.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+This file holds the text to accompany the [What is InnerSource?](https://www.safaribooksonline.com/videos/introduction-to-innersource/9781492041504/9781492041504-video321608) video.
+This accompanying text will appear as a part of the learning path (like [this example](https://www.safaribooksonline.com/learning-paths/learning-path-lean/9781491999738/9781491946527-/part01ch01.html)).
+
+# What is InnerSource?
+
+The basic idea of InnerSource is this: If a consuming team can't get what it needs by submitting a feature request then it may submit a pull request instead.
+The producing team partners to review and accept the pull request and the functionality it implements.
+In this context of a code submission, we call the consuming team the **Guest** team and the producing team the **Host** team.
```
Comment (gruetter) [inline introduction/what-is.md]: Hmm ... I think the terminology is right, but some readers might get confused here, because the **Guest** team actually created the PR and thus _produced_ code, rather than the **Host** team, who, in the best case, just _consumed_ the PR. Maybe there's a way to frame this differently?!

File: introduction/what-is.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+The terms **Guest** and **Host** suggest a situation analogous to receiving a visitor in the home.
```
Comment (gruetter) [inline introduction/what-is.md]: This is a very nice explanation of the _Host/Guest_ analogy. Kudos, @rrrutledge !

File: introduction/what-is.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+There may be a few house rules that guests are asked to follow.
+Similarly, most guests want to show respect for the home and their host.
+They're careful with the items in the house and follow the rules for the duration of their stay.
+They may hope for or expect a return invitation as long as they've been courteous and polite.
+These concepts around a home visit are a metaphor for the attitude and behaviors that teams should bring as one hosts another making a guest contribution to the codebase.
+
+Let's take a closer look at how the mechanics of the inner source process can work.
+To aid in this explanation, we'll name a few key individuals on the guest and host teams.
+First, the **Product Owner** determines what functionality the host team is willing to accept as a contribution.
+The **Contributor** is the individual that submits the code contribution.
```
Comment (gruetter) [inline introduction/what-is.md]: perhaps mention that the contributor is the _guest_ in the _host/guest_ analogy?!

File: introduction/what-is.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+The **Trusted Committer** provides any timely support and mentorship that the contributor needs to successfully submit the pull request.
+
+With those definitions, here is the basic outline of an inner source contribution.
+
+* Guest team requests a feature from the host team.
```
Comment (gruetter) [inline introduction/what-is.md]: This paragraph describes _one_, possibly the most common application of InnerSource. But it's not the only one. Should we also mention that sometimes people just contribute an improvement on their own initiative, like in Open Source? Perhaps we could lead into this paragraph by referring to the example in the last article.

File: introduction/what-is.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+* Product owner ensures that user stories representing the feature request are created.
+These stories should describe the requested feature in terms agreeable to the guest team.
+They also list any details from the host team on how the feature should be delivered in order for the work to be accepted.
+Examples of such details include architecture constraints, coding conventions, dependency usages, data contracts, etc.
+* Supported by the trusted committer, the contributor submits the pull request to implement the requested feature.
+
+Note that these steps do not assume a specific system for the general organization of a team's time or priorities. Inner source assumes that teams already have existing such methods of organization and provides a framework of how to use them to work together where there is a guest team desiring to contribute code to a host.
```
Comment (gruetter) [inline introduction/what-is.md]: s/existing such/existing or s/existing such/such

File: introduction/what-is.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+This option works well for the guest team because they get the functionality they need when they need it without taking on the long-term burden of maintenance of the solution.
```
Comment (gruetter) [inline introduction/what-is.md]: It only works if the _Trusted Committers_ do their job _and_ do it in a **timely fashion**, as you mentioned above. If they don't, the guest team would be back on square one. Maybe reference the trusted committer article here (instead of in the section _Additional Resources_, below.

File: introduction/what-is.md (hunk 3 above)
Comment (KathrynKLewis) [reply introduction/what-is.md]: Agreed @gruetter 
First, as a member of the host team, the **Product Owner** determines what functionality the host team will accept as a contribution. 
Second, the **Contributor**, as a guest, is the individual that submits their code contribution for review by the host.

File: introduction/what-is.md (hunk 4 above)
Comment (rrrutledge) [reply introduction/what-is.md]: I'm not sure.  There isn't any reference here to motivations for _why_ the **Contributor** wants to get involved.  Doesn't what's written here apply equally to those contributing of their own free will just as much as those contributing as part of getting in a dependency for planned work?

File: introduction/what-is.md (hunk 6 above)
Comment (rrrutledge) [reply introduction/what-is.md]: I agree that everyone doing their job is critical to making this process work.  Product Owners, too.  In general we'd like for all kinds of links to exist between these articles.  I'm not exactly sure how to do it.  I think that we'd want for the articles on O'Reilly to link to each other and the articles that we host on _innersourcecommons_ to link to each other.

I'm thinking the best thing  to do here is just remove the **Additional References** section and put a ticket in our _Kanban_ board to go and add in everything linking to each other after we're done producing it all.  What do you think?

File: introduction/what-is.md (hunk 1 above)
Comment (rrrutledge) [reply introduction/what-is.md]: Updated to _consuming_  and _producing_ to _submitting_ and receiving_.

File: introduction/what-is.md (hunk 3 above)
Comment (rrrutledge) [reply introduction/what-is.md]: Sprinkled in references to **Guest** and **Host** teams throughout all three role definitions.

File: introduction/what-is.md (hunk 4 above)
Comment (rrrutledge) [reply introduction/what-is.md]: Made a small update to specify that a team or an contributor may be driving the ask for a change in source code.

File: introduction/what-is.md (hunk 5 above)
Comment (rrrutledge) [reply introduction/what-is.md]: I wish that _GitHub_ had a button that could execute regex ...

File: introduction/what-is.md (hunk 6 above)
Comment (rrrutledge) [reply introduction/what-is.md]: I'm thinking the best thing  to do here is just remove the **Additional References** section and put a ticket in our _Kanban_ board to go and add in everything linking to each other after we're done producing it all.  What do you think?

File: introduction/what-is.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+This file holds the text to accompany the [What is InnerSource?](https://www.safaribooksonline.com/videos/introduction-to-innersource/9781492041504/9781492041504-video321608) video.
+This accompanying text will appear as a part of the learning path (like [this example](https://www.safaribooksonline.com/learning-paths/learning-path-lean/9781491999738/9781491946527-/part01ch01.html)).
+
+# What is InnerSource?
+
+The basic idea of InnerSource is this: If a consuming team can't get what it needs by submitting a feature request then it may submit a pull request instead.
```
Comment (gruetter) [inline introduction/what-is.md]: Proposal: `s/a consuming team/a team using InnerSource software`. Or maybe even use _team A_ and _team B_ to avoid misunderstandings. Here's an attempt:

> Let's say team A is using software produced by team B. Team A submitted a feature request to team B but team B isn't able to implement that feature in time for team A. In an InnerSource setting, team A would then implement the feature itself and submit a Pull Request to team B, which would then partner to review and accept the pull request and thus add the feature to their software. In this example, we call team A the **Guest** team and team B the **Host** team.

File: introduction/what-is.md (hunk 6 above)
Comment (gruetter) [reply introduction/what-is.md]: good idea, @rrrutledge

File: introduction/what-is.md (hunk 7 above)
Comment (rrrutledge) [reply introduction/what-is.md]: Updated.  Have a look.

File: introduction/what-is.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+Similarly, most guests want to show respect for the home and their host.
+They're careful with the items in the house and follow the rules for the duration of their stay.
+They may hope for or expect a return invitation as long as they've been courteous and polite.
+These concepts around a home visit are a metaphor for the attitude and behaviors that teams should bring as one hosts another making a guest contribution to the codebase.
+
+Let's take a closer look at how the mechanics of the inner source process can work.
+To aid in this explanation, we'll name a few key individuals on the guest and host teams.
+First, the **Product Owner** determines what functionality the host team is willing to accept as a contribution.
+The **Contributor** is the individual on the guest team that submits the code contribution for review by the host team.
+The **Trusted Committer** represents the host team in providing any timely support and mentorship that the contributor needs to successfully submit the pull request.
```
Comment (demianbrecht) [inline introduction/what-is.md]: Given our discussion last meeting, could this potentially be a place to hint at the meritocracy model? Something to the effect of "respresents the host team or an external contributor who has been promoted by the host team due to ongoing contributions"? I'm kind of on the fence about it because the meritocracy model hasn't been introduced yet and it feels a little wordy. Just thought I'd throw the thought out there.

File: introduction/what-is.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+With those definitions, here is the basic outline of an inner source contribution.
+
+* Guest team or contributor requests a feature from the host team.
+* Product owner ensures that user stories representing the feature request are created.
+These stories should describe the requested feature in terms agreeable to the guest team.
+They also list any details from the host team on how the feature should be delivered in order for the work to be accepted.
+Examples of such details include architecture constraints, coding conventions, dependency usages, data contracts, etc.
+* Supported by the trusted committer, the contributor submits the pull request to implement the requested feature.
+
+Note that these steps do not assume a specific system for the general organization of a team's time or priorities. Inner source assumes that teams already have existing methods of organization and provides a framework of how to use them to work together where there is a guest team desiring to contribute code to a host.
```
Comment (demianbrecht) [inline introduction/what-is.md]: Should we perhaps have a section somewhere (if it doesn't exist already) that makes some suggestions about such systems?

Comment (rrrutledge) [reply introduction/what-is.md]: I was just thinking of Agile/XP/SAFe/etc.  Don't think we need to call that out.  Just wanted to let people know that you can keep whatever planning and work organization process you used to have and still use inner source.

What do you think?

File: introduction/what-is.md (hunk 8 above)
Comment (lenucksi) [inline introduction/what-is.md]:
```suggestion
The **Trusted Committer** represents the host team in providing any timely support and mentorship that the contributor needs to successfully submit the pull request.

Each role has it's own specifics and there's a lot more detail to it. We've made an extra section for each of them. Click their names to jump to them and learn more now. We encourage you to finish this text to get the big picture though. 
```

File: introduction/what-is.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+This option works well for the guest team because they get the functionality they need when they need it without taking on the long-term burden of maintenance of the solution.
+It works for the host team because they are able to better scale and serve their consumers.
+It works for the company at-large because solutions to shared problems end up in shared, centrally-maintained locations where anyone can use them.
+More engineering time stays focused on producing code that solves company problems rather than the mechanics of the feature negotiation and escalation process.
+
+## Additional Resources
+
+* The [Trusted Committer](https://github.com/paypal/InnerSourcePatterns/blob/master/project-roles/trusted-committer.md) pattern.
```
Comment (lenucksi) [inline introduction/what-is.md]: This learning path lives in the InnerSourceCommons namespace. We're linking to the paypal namespace for our pattern repository. Are there any consideration to actually move that repo over to the new name space?

File: introduction/what-is.md (hunk 6 above)
Comment (lenucksi) [reply introduction/what-is.md]: I'm very much for that idea too. 
Three things: 
- We're waiting for O'Reilly to come back to us with with how the like to have links annotated. I'd would hope that they like the Markdown annotation way ```[this](url)``` - if they don't have any preferences, I propose to suggest that to them.
- We still need to add all the hyperlinks we want then. Added a ticket for that since we probably would need to go through all the files once we know how to do it. See #41.
- The case of expectation management of collaboration through time zones that @gruetter mentioned is a case I  can confirm. It should be mentioned somewhere - probably both sides shortly of this effect on the TC and contributor segment - but I'm more for keeping it out of here.

File: introduction/what-is.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+# What is InnerSource?
+
+Let's say that team A uses software produced by team B.
+Team A submits a feature request to team B, but team B isn't able to implement that feature in time for team A.
+In an InnerSource setting, if team A can't get this feature request then it submits a pull request instead.
+That is to say team A implements the feature directly in team B's software and submits a pull request with the code changes.
+Team B partners to review and accept the submitted code.
+
+In this example, we call team A the **Guest** team and team B the **Host** team.
```
Comment (lenucksi) [inline introduction/what-is.md]:
```suggestion
## Host- and Guest-Team

In this example, we call team A the **Guest** team and team B the **Host** team.
```

File: introduction/what-is.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+They ensure that things are kept neat and tidy in anticipation of their guests' arrival.
+Visitors are greeted at the door and invited in.
+They can use the features and utilities that are in the home's public areas.
+There may be a few house rules that guests are asked to follow.
+Similarly, most guests want to show respect for the home and their host.
+They're careful with the items in the house and follow the rules for the duration of their stay.
+They may hope for or expect a return invitation as long as they've been courteous and polite.
+These concepts around a home visit are a metaphor for the attitude and behaviors that teams should bring as one hosts another making a guest contribution to the codebase.
+
+Let's take a closer look at how the mechanics of the inner source process can work.
```
Comment (lenucksi) [inline introduction/what-is.md]:
```suggestion
## Mechanics
Let's take a closer look at how the mechanics of the inner source process can work.
```

File: introduction/what-is.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+To aid in this explanation, we'll name a few key individuals on the guest and host teams.
+First, the **Product Owner** determines what functionality the host team is willing to accept as a contribution.
+The **Contributor** is the individual on the guest team that submits the code contribution for review by the host team.
+The **Trusted Committer** represents the host team in providing any timely support and mentorship that the contributor needs to successfully submit the pull request.
+
+With those definitions, here is the basic outline of an inner source contribution.
```
Comment (lenucksi) [inline introduction/what-is.md]:
```suggestion
## Outline of a contribution
With those definitions, here is the basic outline of an inner source contribution.
```

File: introduction/what-is.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+* Guest team or contributor requests a feature from the host team.
+* Product owner ensures that user stories representing the feature request are created.
+These stories should describe the requested feature in terms agreeable to the guest team.
+They also list any details from the host team on how the feature should be delivered in order for the work to be accepted.
+Examples of such details include architecture constraints, coding conventions, dependency usages, data contracts, etc.
+* Supported by the trusted committer, the contributor submits the pull request to implement the requested feature.
+
+Note that these steps do not assume a specific system for the general organization of a team's time or priorities. Inner source assumes that teams already have existing methods of organization and provides a framework of how to use them to work together where there is a guest team desiring to contribute code to a host.
+
+This option works well for the guest team because they get the functionality they need when they need it without taking on the long-term burden of maintenance of the solution.
```
Comment (lenucksi) [inline introduction/what-is.md]:
```suggestion
## Realized advantages
This option works well for the guest team because they get the functionality they need when they need it without taking on the long-term burden of maintenance of the solution.
```

File: introduction/what-is.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+They can use the features and utilities that are in the home's public areas.
+There may be a few house rules that guests are asked to follow.
+Similarly, most guests want to show respect for the home and their host.
+They're careful with the items in the house and follow the rules for the duration of their stay.
+They may hope for or expect a return invitation as long as they've been courteous and polite.
+These concepts around a home visit are a metaphor for the attitude and behaviors that teams should bring as one hosts another making a guest contribution to the codebase.
+
+Let's take a closer look at how the mechanics of the inner source process can work.
+To aid in this explanation, we'll name a few key individuals on the guest and host teams.
+First, the **Product Owner** determines what functionality the host team is willing to accept as a contribution.
```
Comment (lenucksi) [inline introduction/what-is.md]: - Danese mentions an interaction of both teams' POs in the video if I remember correctly. I agree with what is written here, but there's a bit of an ambiguity here.
- I was wondering if it would be a good idea to mention that PO and TC can be same person. This is likely not that uncommon with the usual grass-roots style emerging local library/tool/automation/... projects. Other more "planned" projects might have dedicated PO's from the start, however this does not have to be the case - matching superior topic: governance.

File: introduction/what-is.md [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+The **Contributor** is the individual on the guest team that submits the code contribution for review by the host team.
+The **Trusted Committer** represents the host team in providing any timely support and mentorship that the contributor needs to successfully submit the pull request.
+
+With those definitions, here is the basic outline of an inner source contribution.
+
+* Guest team or contributor requests a feature from the host team.
+* Product owner ensures that user stories representing the feature request are created.
```
Comment (lenucksi) [inline introduction/what-is.md]: With the more grass-roots style projects: I would argue that guests might file tickets themselves, get a bunch of feedback around filing it and start bringing up the matching PR's. 
The sentence could be understood as in "ensures that it gets done either by host or guest" or "hosts create them". Could we nudge the sentence a bit more into the "can be created by either side" direction?
```suggestion
* Product owner ensures that user stories representing the feature request are created, either by members of the guest team or host team.
```

File: introduction/what-is.md (hunk 15 above)
Comment (rrrutledge) [reply introduction/what-is.md]: This subject keeps coming up - updated.

File: introduction/what-is.md (hunk 8 above)
Comment (rrrutledge) [reply introduction/what-is.md]: I'm wondering if we can just link the names to the other segments and the UI will make it obvious that it's a link without us needing to add any text telling them to click it.

File: introduction/what-is.md (hunk 10 above)
Comment (rrrutledge) [reply introduction/what-is.md]: I don't think that anyone's brought it up, but I'm sure that folks would be open to it.

File: introduction/what-is.md (hunk 8 above)
Comment (lenucksi) [reply introduction/what-is.md]: That sounds like the right thing to do, let's do it.

File: introduction/what-is.md (hunk 10 above)
Comment (lenucksi) [reply introduction/what-is.md]: I think we should then ask the folks about it - I'll bring it up on both matching channels in Slack.

File: introduction/what-is.md (hunk 15 above)
Comment (lenucksi) [reply introduction/what-is.md]: :+1:

**Comment by lenucksi** [comment]
> I think that the headings do a nice job of summarizing the content of each section, but most of the proposed sections are only 1 paragraph long. Do single paragraphs need section headings? Normally I'd think of those being used to group multiple, related paragraphs together. Let me know what you think.

I've had a second look at the rendered Markdown. I guess it's short enough to not need separate headings.

File: introduction/what-is.md (hunk 8 above)
Comment (lenucksi) [reply introduction/what-is.md]: Now that we can likely hyperlink our other segments, and given that we have (at least I think that we had) something on the "how-to-become a TC" in the TC segment, would you think that this could resolve this conversation @demianbrecht?

**Comment by rrrutledge** [comment]
@lenucksi can you summarize what work you feel is left on this PR?  There's been so many conversations and I think most have ended but let me know if we've missed any?

File: introduction/what-is.md (hunk 9 above)
Comment (demianbrecht) [reply introduction/what-is.md]: Yeah it makes sense. I think how inner sourcing is implemented will be company-specific so makes more sense to keep it abstract.

**Comment by lenucksi** [comment]
> @lenucksi can you summarize what work you feel is left on this PR? There's been so many conversations and I think most have ended but let me know if we've missed any?

I was actually close to merging it, here's a summary:
- There's the point that the title of the section might not be the most perfect (although the title slide of the video actually has the same title). If everyone's good with the current title, I'm happy to go with it too. https://github.com/InnerSourceCommons/InnerSourceLearningPath/pull/6#discussion_r256076640
- Some minor wording around interpretation of a sentence: https://github.com/InnerSourceCommons/InnerSourceLearningPath/pull/6#discussion_r257313446 (I'd be okay with closing it, too.)
- The case of exchanging the link to the patterns repo: https://github.com/InnerSourceCommons/InnerSourceLearningPath/pull/6#discussion_r257845863 We could merge that in now and change it later. Would need to have an issue for that then.

**Comment by rrrutledge** [comment]
* I agree that **How Does InnerSource Work?** would be a better title.  The one caveat here is that we'd need to make sure that O'Reilly can update the intro slide to match.  They made that - we don't have the underlying slides available to us.  I added it to tomorrow's agenda.
* Sounds good.  Updated.
* I think that we should just open an issue to track if we're otherwise ready to merge.

**Comment by lenucksi** [comment]
>     * I agree that **How Does InnerSource Work?** would be a better title.  The one caveat here is that we'd need to make sure that O'Reilly can update the intro slide to match.  They made that - we don't have the underlying slides available to us.  I added it to tomorrow's agenda.

Cool, I'll leave that open until then, such that we can either open a new PR on that or add it to this one. 

>     * I think that we should just open an issue to track if we're otherwise ready to merge.

#45's the matching issue for it, I'll resolve that comment.

File: introduction/what-is.md (hunk 10 above)
Comment (lenucksi) [reply introduction/what-is.md]: We'll track that further in #45.

**Comment by rrrutledge** [comment]
Thanks, @lenucksi.
